eng/versioning/scan_for_unreleased_dependencies.ps1

```diff
@@ -9,7 +9,7 @@ param(
 
 # Given an input groupId, artifactId and root service directory, scan the service directory for the
 # POM file that matches the group/artifact. If the POM file is found, scan for any unreleased_ dependency
-# tags otherwise report an error. If there are unreleased dependency tags then report them and return an 
+# tags otherwise report an error. If there are unreleased dependency tags then report them and return an
```

Comment:
[This](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/storage/azure-storage-blob-cryptography/pom.xml#L112) library already has such a case. Should we be re-releasing then?

cc: @gapra-msft

Reply:
@samvaity that's a question for @scbedd but I believe the answer is no. Because that library was already released and breaking the doc build pipeline is why I'm making this change, to prevent next month's releases from doing the same thing.

Reply:
`beta_` should be an acceptable version tag for release still, right? This basically enforces the pattern where we will prevent a library from shipping using an unreleased version that may never be shipped to Maven. Ex, KeyVault has a current dev version of `-beta.4` but next release will be the next minor version, we'll never go back into beta for a GA'd minor version.